web-api/migration-terraform/main/lambdas/migrations/0025-add-contacts-to-petitioners-array.js

```diff
@@ -0,0 +1,49 @@
+const createApplicationContext = require('../../../../src/applicationContext');
+const {
+  Case,
+} = require('../../../../../shared/src/business/entities/cases/Case');
+const {
+  CONTACT_TYPES,
+} = require('../../../../../shared/src/business/entities/EntityConstants');
+const applicationContext = createApplicationContext({});
+
+const migrateItems = async items => {
+  const itemsAfter = [];
+  for (const item of items) {
+    if (item.pk.startsWith('case|') && item.sk.startsWith('case|')) {
+      item.otherFilers?.forEach(
+        filer => (filer.contactType = CONTACT_TYPES.otherFiler),
+      );
+      item.otherPetitioners?.forEach(
+        filer => (filer.contactType = CONTACT_TYPES.otherPetitioner),
+      );
+
+      item.petitioners = [
+        ...(item.petitioners || []),
+        ...(item.otherFilers || []),
+        ...(item.otherPetitioners || []),
+      ];
+
+      const updatedCaseRaw = new Case(item, { applicationContext })
+        .validate()
+        .toRawObject();
```

Comment:
This migration script is incredibly tidy and easy to follow. One thing I can't quite tell (without digging back into Case) is precisely how the primary and secondary petitions get their CONTACT_TYPES associated with them? I think that's in the constructor, right?

Reply:
... seems well covered in the tests below. 😄  👍